vmware/ngc/NGC-Register/src/main/java/org/opensds/vmware/ngc/utils/XmlUtils.java

```diff
@@ -0,0 +1,350 @@
+        if (OperationSystemUtils.isFreeDiskSpace()) {
+            return ErrorCode.NO_SPACE;
+        }
+        OutputStream outputStream = null;
+        Writer writer = null;
+        XMLWriter xmlWriter = null;
+        try {
+            outputStream = new FileOutputStream(path);
+            writer = new OutputStreamWriter(outputStream, "UTF-8");
+            xmlWriter = new XMLWriter(writer);
+            xmlWriter.write(doc);
+            xmlWriter.flush();
+            return ErrorCode.SUCCESS;
+        } catch (IOException e) {
+            _logger.error("Writer xml error ;" + e.toString());
+            return ErrorCode.WRITE_CONFIG_FAILED;
+        } finally {
+            if (xmlWriter != null) {
+                try {
+                    xmlWriter.close();
+                }
+                catch (IOException e) {
+                    _logger.error("XMLWriter close error" + e.toString());
+                }
+            }
+            closeStream(writer);
+            closeStream(outputStream);
+        }
+    }
+
+    public Document loadConigXml(String filename) {
+        Document document = null;
+        Reader xmlReader = null;
+        FileInputStream stream = null;
+        try {
+            SAXReader saxReader = new SAXReader();
+            stream =  new FileInputStream(new File(filename));
+            xmlReader = new InputStreamReader(stream, "UTF-8");
+            document = saxReader.read(xmlReader);
+        }
+        catch (Exception ex) {
+            _logger.error(String.format("the filename of document is %s, error [%s]" , filename
+                    ,ex.getMessage()));
+            document = DocumentHelper.createDocument();
+            document.addElement("ROOT");
+        } finally {
+            closeStream(xmlReader);
+            closeStream(stream);
+        }
+        return document;
+    }
+
+
+    public Document load(String filename) {
+        Document document = null;
+        Reader xmlReader = null;
+        InputStream stream = null;
+        try {
+            SAXReader saxReader = new SAXReader();
+            stream = getClass().getClassLoader().getResourceAsStream(filename);
+            xmlReader = new InputStreamReader(stream, "UTF-8");
+            document = saxReader.read(xmlReader);
+        }
+        catch (Exception ex) {
+            _logger.error(String.format("the filename of document is %s, error [%s]" , filename
+                    ,ex.getMessage()));
+            document = DocumentHelper.createDocument();
+            document.addElement("ROOT");
+        } finally {
+            closeStream(xmlReader);
+            closeStream(stream);
+        }
+        return document;
+    }
+
+    private void closeStream(Closeable stream) {
+        if (stream != null) {
+            try {
+                stream.close();
+            }
+            catch (IOException e) {
+                _logger.error(e.getMessage(), e);
```

Comment:
why no error code?

Reply:
closeStream is a void method. That's why not returning any error code.